test/In.ProjectEKA.HipServiceTest/DataFlow/EncryptorTest.cs

```diff
@@ -0,0 +1,79 @@
+namespace In.ProjectEKA.HipServiceTest.DataFlow
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Bogus;
+    using FluentAssertions;
+    using HipLibrary.Patient.Model;
+    using HipService.DataFlow.Encryptor;
+    using Moq;
+    using Xunit;
+    using Encoder = Org.BouncyCastle.Utilities.Encoders;
+
+    public class EncryptorTest
+    {
+        private readonly string curve = "curve25519";
+        private readonly string algorithm = "ECDH";
+        private readonly Encryptor encryptor = new Encryptor();
+        [Fact]
+        private void ShouldGenerateKeyPair()
+        {
+            var keyPair = EncryptorHelper.GenerateKeyPair(curve, algorithm);
+            keyPair.Should().NotBeNull();
+        }
+
+        [Fact]
+        private void ShouldReturnPublicKey()
+        {
+            var keyPair = EncryptorHelper.GenerateKeyPair(curve, algorithm);
+            var publicKey = EncryptorHelper.GetPublicKey(keyPair);
+            publicKey.Should().NotBeNull();
+        }
+
+        [Fact]
+        private void ShouldGenerate32ByteRandomKey()
+        {
+            var randomKey = EncryptorHelper.GenerateRandomKey();
+            var randomKeyByte = GetByteFromBase64(randomKey);
+
+            randomKeyByte.Count().Should().Be(32);
+        }
+
+        [Fact]
+        private void ShouldEncryptData()
+        {
+            var randomKeyFirst = EncryptorHelper.GenerateRandomKey();
+            var randomKeySecond = EncryptorHelper.GenerateRandomKey();
+            var senderKeyPair = EncryptorHelper.GenerateKeyPair(curve, algorithm);
+            var receiverPublicKey =
+                "BE7rHCAB0xdU/wTh4PeZhnmLfPSS2CQQo3Loi2D3sWiGBwm9lwsxudMgTyjZ0KUzEqUniuW7zmDr3Vy9JeYI7gA=";
+            var keyMaterial = new KeyMaterial(algorithm, curve, new KeyStructure(It.IsAny<string>(), It.IsAny<string>(), receiverPublicKey), randomKeyFirst);
+
+            var result = encryptor.EncryptData(keyMaterial, senderKeyPair, new Faker().Random.Word(), randomKeySecond);
+
+            result.HasValue.Should().BeTrue();
+            result.MatchSome(data => data.Length.Should().BePositive());
+        }
+        
+        [Fact]
+        private void ShouldNotEncryptData()
+        {
+            var faker = new Faker();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'faker'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5073772)